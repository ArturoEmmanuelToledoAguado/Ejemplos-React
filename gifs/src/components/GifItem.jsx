import PropTypes from 'prop-types';

export const GifItem = ({title,url}) => {
    const copyPath = ()=>{
        navigator.clipboard.writeText(url)
    }

  return (
    <div className="">
        <img 
            src={url}
            alt={title}
        />
        <p>{title}</p>
        <button className=""onClick={copyPath}>Copy Path</button>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}