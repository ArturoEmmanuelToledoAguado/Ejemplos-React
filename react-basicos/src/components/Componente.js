import React from "react";
//import React,{Component} from "react";

const Componente=props=> <h2>{props.msg}</h2>

/* Componente en funcion declarada
function Componente(props) {
    return <h2>{props.msg}</h2>;
}*/

/* Componente con clase
class Componente extends Component{
    render() {
        return <h2>{this.props.msg}</h2>;
    }
}*/

export default Componente;