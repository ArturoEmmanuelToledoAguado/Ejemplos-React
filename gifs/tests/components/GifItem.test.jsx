import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('prueba sobre GifItem', () => { 
    const title='batman'
    const url = 'http://github.com/'
    test('Esta prueba es para hacer match con el snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
        //screen.debug()
     })
     
    test('Validar la existencia de las imagenes', () => { 
        render(<GifItem title={title} url={url}/>)
        const {alt,src} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toEqual(title)
      })

    test('Validar la existencia del boton', () => { 
        render(<GifItem title={title} url={url}/>)
        const {className} = screen.getByRole('button')
        expect(className).toBe('')
        screen.debug()
     })

    /* test('Valida la existencia del div', () => { 
        render(<GifItem title={title} url={url}/>)
        const {className} = screen.getByRole('div')
        expect(className).toBe('')
        screen.debug()
     }) */
 })