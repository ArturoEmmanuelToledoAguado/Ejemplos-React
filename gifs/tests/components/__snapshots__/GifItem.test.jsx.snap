// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba sobre GifItem Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <div
    class=""
  >
    <img
      alt="batman"
      src="http://github.com/"
    />
    <p>
      batman
    </p>
    <button
      class=""
    >
      Copy Path
    </button>
  </div>
</div>
`;
